File: examples/governance_static_go/domain/resolved_incident.go
package domain

import "time"

type ResolvedIncident struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	IncidentId string `json:"incident_id"`
	Resolution string `json:"resolution"`
	RootCause string `json:"root_cause"`
	ModelId string `json:"model_id"`
	Severity string `json:"severity"`
	Category string `json:"category"`
	Description string `json:"description"`
	ReportedById string `json:"reported_by_id"`
	ReportedAt time.Time `json:"reported_at"`
	ResolvedAt time.Time `json:"resolved_at"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e ResolvedIncident) EventName() string { return "ResolvedIncident" }

func (e ResolvedIncident) GetOccurredAt() time.Time { return e.OccurredAt }

File: examples/governance/operations_domain/lib/operations_domain/incident/events/resolved_incident.rb
module OperationsDomain
  class Incident
    module Events
      class ResolvedIncident
        attr_reader :aggregate_id, :incident_id, :resolution, :root_cause, :model_id, :severity, :category, :description, :reported_by_id, :reported_at, :resolved_at, :status, :occurred_at

        def initialize(aggregate_id: nil, incident_id: nil, resolution: nil, root_cause: nil, model_id: nil, severity: nil, category: nil, description: nil, reported_by_id: nil, reported_at: nil, resolved_at: nil, status: nil)
          @aggregate_id = aggregate_id
          @incident_id = incident_id
          @resolution = resolution
          @root_cause = root_cause
          @model_id = model_id
          @severity = severity
          @category = category
          @description = description
          @reported_by_id = reported_by_id
          @reported_at = reported_at
          @resolved_at = resolved_at
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end
